### PR TITLE
Change Schema Snapshot Sort Ordering of Fields/Relations

### LINKED ISSUE #19756: Snapshot field ordering is wrong
### Describe the Bug

While trying to apply an SQLite snapshot to Postgres found out that columns are being created in a random order, making it impossible to use 'INSERT INTO' as the values are incorrect

### To Reproduce

on a sqlite backed directus
`npx directus schema snapshot ./snapshot.yaml`

on a new bootstrapped Postgres backed directus
`npx directus schema apply ./snapshot.yaml`

Export data and import through pgql
`sqlite3 data.db .dump | grep '^INSERT INTO' > dump.sql`

Also, tried to convert data by adding the table order on the statements though, this is still a work in progress:
```python
# python
import sqlite3

def escape_quotes(value):
    """Escape single and double quotes for SQL insertion."""
    if isinstance(value, str):
        return value.replace("'", "''").replace('"', '""')
    return value

db_file = 'data.db'
conn = sqlite3.connect(db_file)
cursor = conn.cursor()

tables = [row[0] for row in cursor.execute("SELECT name FROM sqlite_master WHERE type='table';").fetchall()]

with open("dump.sql", "w") as out_file:
    for table in tables:
        # Get column names for the table
        columns = [row[1] for row in cursor.execute(f"PRAGMA table_info({table})").fetchall()]

        # Fetch data from the table
        data_rows = cursor.execute(f"SELECT * FROM {table}").fetchall()

        # Write data to the file
        for row in data_rows:
            values = ', '.join([f"'{escape_quotes(value)}'" if isinstance(value, str) else str(value) for value in row])
            out_file.write(f"INSERT INTO {table} ({', '.join(columns)}) VALUES ({values});\n")

cursor.close()
conn.close()
```

### Directus Version

v10.6.2

### Hosting Strategy

Self-Hosted (Custom)

### PR DESCRIPTION
**Pull request by connorwinston**
## Scope

What's changed:

- Changed the ordering of **fields** in `getSnapshot` from `collection,field` to `meta.id`
- Changed the ordering of **relations** in `getSnapshot` from `collection,field` to `meta.id`

## Potential Risks / Drawbacks

- Changes the sort order of all schema's

## Review Notes / Questions

- Is this considered a breaking change?

---

Fixes #19756

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
The new order will work the same way as the old order right? Or will it start updating fields it shouldn't because the order of the file is wrong 🤔 If it remains the same but the file is exported in a different order once I don't see that as a breaking change myself

**Comment by connorwinston** [comment]
> The new order will work the same way as the old order right? Or will it start updating fields it shouldn't because the order of the file is wrong 🤔 If it remains the same but the file is exported in a different order once I don't see that as a breaking change myself

It shouldn't because each collection, field, and relation are diffed individually not the entire schema as a whole. From my testing it doesn't change anything.

File: .changeset/witty-carpets-add.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@directus/api": patch
+---
+
+Changed Schema getSnapshot Sort Ordering of Fields/Relations to creation order
```
Comment (paescuj) [inline .changeset/witty-carpets-add.md:5]:
```suggestion
Changed schema snapshot sort ordering of Fields/Relations to creation order
```

File: api/src/utils/get-snapshot.ts [hunk 2]
```diff
@@ -26,12 +26,12 @@ export async function getSnapshot(options?: { database?: Knex; schema?: SchemaOv
 	]);
 
 	const collectionsFiltered = collectionsRaw.filter((item: any) => excludeSystem(item));
-	const fieldsFiltered = fieldsRaw.filter((item: any) => excludeSystem(item)).map(omitID) as SnapshotField[];
-	const relationsFiltered = relationsRaw.filter((item: any) => excludeSystem(item)).map(omitID) as SnapshotRelation[];
+	const fieldsFiltered = fieldsRaw.filter((item: any) => excludeSystem(item)) as SnapshotField[];
+	const relationsFiltered = relationsRaw.filter((item: any) => excludeSystem(item)) as SnapshotRelation[];
```
Comment (paescuj) [inline api/src/utils/get-snapshot.ts:30]: Assertion no longer required due to relocation of `map`
```suggestion
	const fieldsFiltered = fieldsRaw.filter((item: any) => excludeSystem(item));
	const relationsFiltered = relationsRaw.filter((item: any) => excludeSystem(item));
```

**Review by paescuj — APPROVED** [review]
Tested & looks good to me 🚀 

While I agree that it is not a breaking change, we should let the admins know that the order has changed and they might want to regenerate the snapshot after the update, in case it's tracked in a VCS so future changes to the schema won't be polluted.
I've placed my proposal in #20844, feel free to adjust.

**Pull request merged**